guide_drafts/model-derives.md

```diff
@@ -0,0 +1,926 @@
+            ),
+
+            ColumnInsertValue::Expression(
+                users::last_name,
+                AsExpression::<<users::email as Expression>::SqlType>::as_expression(email)
+            ),
+        )
+    }
+}
+
+impl <'a: 'insert, 'insert, Op> IntoInsertStatement<users::table, Op> for &'insert NewUser<'a> {
+    type InsertStatement = InsertStatement<users::table, Self, Op>;
+
+    fn into_insert_statement(self, target: users::table, operator: Op) -> Self::InsertStatement {
+            InsertStatement::no_returning_clause(target, self, operator)
+    }
+}
+```
+
+One interesting piece to note are the values of the tuple for the associated type, `Values`.
+An `InsertValue` is a tuple that consists of each column to be inserted as an `ColumnInsertValue`.
+The returned value of `values()` has been formatted to illustrate this point.
+This implementation is considerably complex and you wouldn't want to be typing this out for each `Insertable` struct.
+
+## <a name="identifiable"></a>Identifiable
+
+The `Identifiable` trait is useful when you need to set up database [Associations](#associations) or are using certain Diesel database update features.
+When using the `Indentifiable` on a struct,
+it means that struct can be identified on a single table in the database. 
+
+Your struct *must* have an `id` field on it.
+By default, `Identifiable` will assume the primary key is a column named `id`.
+If your table's primary key is named differently, you can annotate the table with the attribute `#[primary_key(some_field_name)` or `#[primary_key(some_field_name, another_field_name)`.
+Like Insertable, the `Identifiable` trait will assume that the annotated struct will be named in the singular form of the table it corresponds to.
+If the name differs you may use the `#[table_name="some_table_name"]` attribute annotation.
+Having the `Identifiable` trait provides us the `id()` method on our models,
+which returns the value of our record's primary key.
+
+In the following example,
+we will look at some of the behavior `Identifiable` provides for us.
+We will add the annotation to our `User` struct.
+We will then attempt to get the value of the first record's primary key by calling `id()` and also update the first and last name of our user.
+
+### Example 
+
+*Note: src/lib.rs with #[macro_use] extern crate diesel_codegen; not shown*
+
+*src/models.rs*
+```rust
+#[derive(Identifiable, Queryable)]
+pub struct User {
+    pub id: i32,
+    pub first_name: String,
+    pub last_name: String,
+    pub email: String,
+}
+
+#[derive(Insertable)]
+#[table_name="users"]
+pub struct NewUser<'a> {
+    pub first_name: &'a str,
+    pub last_name: &'a str,
+    pub email: &'a str,
+ }
+```
+
+*src/main.rs*
+```rust
+// externs, database connection, and using statement code omitted
+
+fn main() {
+  let new_user = NewUser { 
+      first_name: "Gordon", 
+      last_name: "Freeman", 
+      email: "gordon.freeman@blackmesa.co" 
+  };
```

Comment:
[now we can actually merge it](https://gitter.im/diesel-rs/diesel?at=5990b9e51c8697534ab4b118)

Reply:
I want to make sure this PR abides by Diesel (unofficial) Contributing docs. :smile: